cmd/edit/cluster/cmd.go

```diff
@@ -1142,3 +1189,163 @@ func confirmMigration() (bool, error) {
+	}
+	return ""
+}
+
+func setAutoNode(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, autonode string, roleArn string) (
+	mode string, role string, err error) {
+
+	// Check if flags are changed
+	autonodeChanged := cmd.Flags().Changed("autonode")
+	roleArnChanged := cmd.Flags().Changed("autonode-iam-role-arn")
+
+	if !autonodeChanged && !roleArnChanged {
+		return "", "", nil
+	}
+
+	// Validate HCP cluster
+	if !aws.IsHostedCP(cluster) {
+		return "", "", fmt.Errorf("AutoNode is only supported for Hosted Control Plane clusters")
+	}
+
+	// Get current state
+	currentAutoNodeEnabled := autonodeExists(cluster)
+	currentRoleArn := autonodeRoleExists(cluster)
+
+	// Validate autonode value
+	if autonodeChanged {
+		if autonode != "enabled" {
+			return "", "", fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+		}
+		mode = autonode
+	}
+
+	// Handle role ARN
+	if roleArnChanged {
+		if roleArn != "" && !aws.RoleArnRE.MatchString(roleArn) {
+			return "", "", fmt.Errorf("Invalid IAM Role ARN format: %s", roleArn)
+		}
+		role = roleArn
+	}
+
+	// If enabling AutoNode, role ARN is required
+	if mode == "enabled" && !currentAutoNodeEnabled {
+		if role == "" && currentRoleArn == "" {
+			return "", "", fmt.Errorf("IAM Role ARN is required when enabling AutoNode. Use --autonode-iam-role-arn to specify the role")
+		}
+		if role == "" {
+			role = currentRoleArn
+		}
+	}
+
+	// If only updating role ARN, AutoNode must already be enabled
+	if roleArnChanged && !autonodeChanged && !currentAutoNodeEnabled {
+		return "", "", fmt.Errorf("Cannot update AutoNode IAM Role ARN when AutoNode is not enabled. Enable AutoNode first using --autonode=enabled")
+	}
+
+	// If updating role ARN with AutoNode already enabled, maintain enabled state
+	if roleArnChanged && currentAutoNodeEnabled && !autonodeChanged {
+		mode = "enabled"
+	}
+
+	// Confirm with user
+	if mode != "" || role != "" {
+		confirmAutoNode(r, mode, role, currentAutoNodeEnabled, currentRoleArn)
+	}
+
+	return mode, role, nil
+}
+
+func confirmAutoNode(r *rosa.Runtime, mode string, roleArn string, currentEnabled bool, currentRoleArn string) {
+	if mode == "enabled" && !currentEnabled {
+		r.Reporter.Warnf("You are choosing to enable AutoNode for automatic node management")
+		if !confirm.Confirm("enable AutoNode for cluster with the provided role arn '%s'", roleArn) {
+			os.Exit(0)
+		}
+	} else if roleArn != "" && currentRoleArn != "" && roleArn != currentRoleArn {
+		r.Reporter.Warnf("You are choosing to update the AutoNode IAM role ARN")
+		if !confirm.Confirm("update AutoNode IAM role ARN from '%s' to '%s'", currentRoleArn, roleArn) {
+			os.Exit(0)
+		}
+	}
+}
+
+func autonodeInteractivePrompt(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster) (
+	mode string, roleArn string, err error) {
+
+	// Only prompt if AutoNode is not already enabled
+	currentAutoNodeEnabled := autonodeExists(cluster)
+	if currentAutoNodeEnabled {
```

Comment:
I don't think we don't need to make this a var since it's only used in the if statement right below it, WDYT?